libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -239,64 +272,164 @@ string IRGenerator::generateInitialAssignment(VariableDeclaration const& _varDec
+		"param_" + m_context.functionName(*_contract.constructor()) + "_",
+		0,
+		_contract.constructor()->parameters().size()
+	);
+
+	Whiskers t(R"X(
+		let programSize := datasize("<object>")
+		let argSize := sub(codesize(), programSize)
 
-			<assignToParams> <abiDecode>(memoryDataOffset, add(memoryDataOffset, argSize))
+		let memoryDataOffset := <allocate>(argSize)
+		codecopy(memoryDataOffset, programSize, argSize)
 
-			<constructorName>(<params>)
-		)X");
-		t("object", creationObjectName(_contract));
-		t("allocate", m_utils.allocationFunction());
-		t("assignToParams", paramVars == 0 ? "" : "let " + suffixedVariableNameList("param_", 0, paramVars) + " := ");
-		t("params", suffixedVariableNameList("param_", 0, paramVars));
-		t("abiDecode", abiFunctions.tupleDecoder(constructor->functionType(false)->parameterTypes(), true));
-		t("constructorName", m_context.functionName(*constructor));
+		<assignToParams> <abiDecode>(memoryDataOffset, add(memoryDataOffset, argSize))
+	)X");
+	t("object", creationObjectName(_contract));
+	t("allocate", m_utils.allocationFunction());
+	t("assignToParams", params.empty() ? "" : "let " + params + " := ");
+	t("abiDecode", abiFunctions.tupleDecoder(_contract.constructor()->functionType(false)->parameterTypes(), true));
+
+	m_baseConstructorParameters[_contract.constructor()] = params;
+
+	return t.render();
+}
+
+string IRGenerator::callConstructor(FunctionDefinition const& _constructor)
+{
+	Whiskers t(R"X(
+		<constructorName>(<params><comma><additionalParams>)
+	)X");
+	t("constructorName", m_context.functionName(_constructor));
+	string params = m_baseConstructorParameters[&_constructor];
+	m_baseConstructorParameters.erase(&_constructor);
+	t("params", params);
+	string additionalParams;
+	for (auto const& baseParams: m_baseConstructorParameters)
+		additionalParams += (additionalParams.empty() ? "" : ", ") + baseParams.second;
+	m_additionalParameters[&_constructor] = additionalParams;
+	t("comma", params.empty() || additionalParams.empty() ? "" : ", ");
+	t("additionalParams", additionalParams);
+
+	return t.render();
+}
+
+string IRGenerator::initStateVarsBaseWithoutConstructor(ContractDefinition const& _contract)
+{
+	using boost::adaptors::reverse;
+
+	vector<ContractDefinition const*> toInitStateVars;
+	for (ContractDefinition const* contract: _contract.annotation().linearizedBaseContracts)
+		if (!contract->constructor())
+			toInitStateVars.push_back(contract);
+		else if (*contract != _contract)
+			break;
+
+	string initStateVariablesCode;
+	for (ContractDefinition const* contract: reverse(toInitStateVars))
+		initStateVariablesCode += initStateVariables(*contract);
+
+	return initStateVariablesCode;
+}
+
+string IRGenerator::constructorCode(ContractDefinition const& _contract)
+{
+	// Initialization of state variables in base-to-derived order.
+	solAssert(!_contract.isLibrary(), "Tried to initialize state variables of library.");
+
+	Whiskers constructorCode(R"(
+		<?hasCallValueCheck>
```

Comment:
I think in this case - just a list of template parameters that replaced by the return value of a function call - using Whiskers does not have a real benefit.